File: LeetCode/ConvertSortedArraytoBinarySearchTree.java
package exercise.LeetCode;

public class ConvertSortedArraytoBinarySearchTree {
    public TreeNode sortedArrayToBST(int[] nums) {
        return dfs(nums, 0, nums.length-1);
    }
    public TreeNode dfs(int[] nums, int left, int right) {
        if(left > right) return null;
        int mid = (left+right) / 2;
        TreeNode root = new TreeNode(nums[mid]);
        root.left = dfs(nums, left, mid-1);
        root.right = dfs(nums, mid+1, right);
        return root;
    }

}

File: LeetCode/ConvertSortedArraytoBinarySearchTree.kt
package exercise.LeetCode

fun main(){

}

fun sortedArrayToBST(nums: IntArray): TreeNode? {
    if (nums.isEmpty())
        return null
    fun createBFS(start:Int,end:Int): TreeNode?{
        if (start>end)
            return null
        val mid=(end-start)/2+start
        val root= TreeNode(nums[mid])
        root.left=createBFS(start,mid-1)
        root.right=createBFS(mid+1,end)
        return root
    }

    return createBFS(0,nums.size-1)

}